Build failure with -Wdefaulted-function-deleted

Because Context has a default move assignment operator but its data member Span explicitly deletes it.

Example build failure:
INFO: From Compiling external/io_opencensus_cpp/opencensus/stats/internal/recording.cc:
Step #4: In file included from external/io_opencensus_cpp/opencensus/stats/internal/recording.cc:21:
Step #4: In file included from external/io_opencensus_cpp/opencensus/tags/context_util.h:18:
Step #4: external/io_opencensus_cpp/opencensus/context/context.h:52:12: warning: explicitly defaulted move assignment operator is implicitly deleted [-Wdefaulted-function-deleted]
Step #4:   Context& operator=(Context&&) = default;
Step #4:            ^
Step #4: external/io_opencensus_cpp/opencensus/context/context.h:76:27: note: move assignment operator of 'Context' is implicitly deleted because field 'span_' has a deleted move assignment operator
Step #4:   opencensus::trace::Span span_;
Step #4:                           ^
Step #4: external/io_opencensus_cpp/opencensus/trace/span.h:121:9: note: 'operator=' has been explicitly marked deleted here
Step #4:   Span& operator=(Span&&) = delete;
Step #4:         ^
Step #4: 1 warning generated.

--- a/opencensus/trace/span.h
+++ b/opencensus/trace/span.h
@@ -118,7 +118,7 @@ class Span final {
   // Spans can be move-constructed, but this is mostly for convenient syntax,
   // not performance reasons.
   Span(Span&&) = default;
-  Span& operator=(Span&&) = delete;
+  Span& operator=(Span&&) = default;
 
   // Attempts to insert an attribute into the Span, unless it already exists in
   // which case it will update the value of that attribute. If the max number of

